### PR TITLE
Add Multi Stage Dockerfile with backend:bundle command

### PR DESCRIPTION
**Pull request by benjdlambert**
## Hey, I just made a Pull Request!
Adding this to contrib for now, but maybe in the future we want to make this the default behaviour when we deprecate the `build-image` command @Rugvip?

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: contrib/docker/multi-stage-with-bundle/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
+
+RUN yarn install --production --network-timeout 600000 && rm -rf "$(yarn cache dir)"
+
+COPY --from=build /app/packages/backend/dist/bundle.tar.gz .
+RUN tar xzf bundle.tar.gz && rm bundle.tar.gz
+
+COPY app-config.yaml app-config.production.yaml ./
+
+CMD ["node", "packages/backend", "--config", "app-config.yaml", "--config", "app-config.production.yaml"]
```
Comment (OrkoHunter) [inline contrib/docker/multi-stage-with-bundle/Dockerfile:42]: Enable new line on save in [VSCode](https://stackoverflow.com/a/44704969/4698026). :)

**Comment by Rugvip** [comment]
Yep, will be replacing the Dockerfile in the app template with one based around `backend: bundle`

File: contrib/docker/multi-stage-with-bundle/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Stage 1 - Create yarn install skeleton layer
+FROM node:14-buster AS packages
+
+WORKDIR /app
+COPY package.json yarn.lock ./
+
+# Uncomment this line if building a non create-app version
+# COPY packages packages
```
Comment (Rugvip) [inline contrib/docker/multi-stage-with-bundle/Dockerfile:8]: Seems inverted, packages is always there, plugins may not be

**Comment by benjdlambert** [comment]
@Rugvip so does it make sense to put this one in the contrib folder for now?

**Comment by Rugvip** [comment]
Tbh maybe we should put it in docs instead

File: contrib/docker/multi-stage-with-bundle/Dockerfile (hunk 2 above)
Comment (benjdlambert) [reply contrib/docker/multi-stage-with-bundle/Dockerfile:8]: derp

**Review by adamdmharvey — COMMENTED** [review]
Great instructions.

Thinking from my perspective two months ago, two points:

* Assume this uses SQLite, is that right?  I wonder maybe just a sentence at the top that sort of describes that, because at first it wasn't quite clear to me that, say, if I shut it down it didn't also contain a database that was persisting. (as I hadn't gotten that far into the code base)  So just maybe a super brief note that this includes an in-memory database that will persist as long as the Docker container is running?
* For newbies, a very last bullet that says something like load your browser on http://localhost:7000? Simple I know, but when doing local `yarn start` dev since the service fires it up, it's super easy for newbies, but just may help be comprehensive. (and I appreciate, if you don't know that maybe running docker locally isn't ideal :) )

**Comment by Rugvip** [comment]
@adamdmharvey The Dockerfile is for both SQLite and PostgreSQL, but the run instructions would ofc only work with sqlite. Perhaps we can mention that that example assumes you're using sqlite and that you'd need a postgres instance to connect to when using postgres.

+1 to mentioning opening browser at localhost:7000

**Comment by benjdlambert** [comment]
@adamdmharvey @Rugvip can I get a hell yeah?

**Review by adamdmharvey — APPROVED** [review]
yeeeeeeah bentopus! 🐙

**Review by jhaals — APPROVED** [review]

**Review by Rugvip — APPROVED** [review]

**Pull request merged**